cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -269,6 +271,7 @@ impl NetworkRunnable for Cmd {
                 );
             }
 
+            let assembled = simulate_if_needed().await?;
```

Comment:
If should send is not yes, and not build only, won't it simulate twice? It looks like it calls that function twice.